dynosaur_derive/src/lib.rs

```diff
@@ -139,40 +139,110 @@ impl Parse for Bridge {
 ///
 /// The `Dyn` struct produced by this macro has the following constructors:
 ///
-/// ```ignore
+/// ```
+/// # struct DynTrait<'a>(&'a i32);
+/// # trait Trait {}
 /// impl<'a> DynTrait<'a> {
-///     fn new_box(from: impl Trait) -> Box<Self> { ... }
-///     fn new_arc(from: impl Trait) -> std::sync::Arc<Self> { ... }
-///     fn new_rc(from: impl Trait) -> std::rc::Rc<Self> { ... }
-///     fn from_ref(from: &'a impl Trait) -> &'a Self { ... }
-///     fn from_mut(from: &'a mut impl Trait) -> &'a mut Self { ... }
+///     fn new_box(from: impl Trait) -> Box<Self> { todo!() }
+///     fn new_arc(from: impl Trait) -> std::sync::Arc<Self> { todo!() }
+///     fn new_rc(from: impl Trait) -> std::rc::Rc<Self> { todo!() }
+///     fn from_ref(from: &'a impl Trait) -> &'a Self { todo!() }
+///     fn from_mut(from: &'a mut impl Trait) -> &'a mut Self { todo!() }
 /// }
 /// ```
 ///
 /// Normally a concrete type behind a pointer coerces to `dyn Trait` implicitly.
 /// When using the `Dyn` struct created by this macro, such conversions must be
 /// done explicitly with the provided constructors.
 ///
+/// ## Bridge impls
+///
+/// Dynosaur writes the following *bridge implementations* for your trait:
+///
+/// ```
+/// # trait Trait {}
+/// // Always:
+/// impl<T: Trait> Trait for Box<T> { }
+/// // If all method receivers are `&mut self` or `&self`:
+/// impl<T: Trait> Trait for &mut T { }
+/// // If all method receivers are `&self`:
+/// impl<T: Trait> Trait for &T { }
+/// ```
+///
+/// This can be controlled with the `bridge` option.
+///
+/// ```
+/// # mod dynosaur { pub use dynosaur_derive::dynosaur; }
```

Comment:
Not sure I understand why you added this module and added a use referring to the internal dynosaur_derive crate. Is this needed?

Reply:
Yeah, it's needed because these docs are in dynosaur_derive, which cannot see the `dynosaur` crate we want people to actually use.